File: meme_inator_back/apps/posts/domain/entities/post_entity.py
from dataclasses import dataclass, field
from datetime import datetime
from typing import List, Optional
from uuid import UUID

@dataclass
class PostEntity:
    post_id: UUID
    imageURL: str
    author: UUID = None
    thumbnailURL: Optional[str] = None
    caption: Optional[str] = None
    createdOn: Optional[datetime] = None
    post_type: Optional[str] = None
    fileFormat: Optional[str] = None
    upvotesCount: int = 0
    downvotesCount: int = 0
    commentsCount: int = 0
    sharesCount: int = 0
    # field(default_factor) prevents each instance sharing the same list object. 
    tags: List[str] = field(default_factory=list) 
    isFlagged: bool = False
    isDeleted: bool = False
    visibility: Optional[str] = None

    def set_image_url(self, url: str):
        self.imageURL = url

    def set_thumbnail_url(self, url: Optional[str]):
        self.thumbnailURL = url

File: meme_inator_front/lib/features/post/domain/entities/post_entity.dart
// apps/posts/domain/entities/post_entity.dart
import 'package:uuid/uuid.dart';

class PostEntity {
  UuidValue postId;
  String imageUrl;
  UuidValue authorId;
  String? thumbnailUrl;
  String? caption;
  DateTime? createdOn;
  String? postType;
  String? fileFormat;

  int likesCount;
  
  int dislikesCount;
  int commentsCount;
  int sharesCount;

  List<String> tags;
  bool isFlagged;
  bool isDeleted;
  String? visibility;

  PostEntity({
    required this.postId,
    required this.imageUrl,
    required this.authorId,
    this.thumbnailUrl,
    this.caption,
    this.createdOn,
    this.postType,
    this.fileFormat,
    this.likesCount = 0,
    this.dislikesCount = 0,
    this.commentsCount = 0,
    this.sharesCount = 0,
    List<String>? tags,
    this.isFlagged = false,
    this.isDeleted = false,
    this.visibility,
  }) : tags = tags ?? [];

  void setImageUrl(String url) => imageUrl = url;
  void setThumbnailUrl(String? url) => thumbnailUrl = url;
}
